etcdserver/server.go

```diff
@@ -204,6 +204,13 @@ func NewServer(cfg *ServerConfig) (*EtcdServer, error) {
 		cl *cluster
 	)
 
+	if mkerr := fileutil.MkdirAllIfNotExist(cfg.DataDir); mkerr != nil {
+		return nil, fmt.Errorf("cannot create data directory: %v", mkerr)
+	}
+	if wderr := fileutil.IsDirWriteable(cfg.DataDir); wderr != nil {
```

Comment:
The `IsDirWriteable` check probably belongs in the `MkdirAllIfNotExist` since every call will probably be followed by the the writeable test otherwise.